nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/AbstractDatabaseFetchProcessor.java

```diff
@@ -436,13 +437,26 @@ protected static String getLiteralByType(int type, String value, String database
             case NVARCHAR:
             case VARCHAR:
             case ROWID:
-            case DATE:
-            case TIME:
                 return "'" + value + "'";
+            case TIME:
```

Comment:
See my comment below, perhaps we'd have three methods then, getDateLiteral(), getTimeLiteral(), and getTimestampLiteral().